qemu/tests/qemu_guest_agent.py

```diff
@@ -1633,6 +1633,66 @@ def check_vss_info(cmd_type, key, expect_value):
                 test.error("guest-fsfreeze-thaw cmd failed with:"
                            "('%s')" % str(detail))
 
+    @error_context.context_aware
+    def gagent_check_fsinfo(self, test, params, env):
+        """
+        Execute "guest-fsinfo" command to guest agent,check file system of
+        mountpoint,disk's name and serial number.
+
+        steps:
+        1) check file system of every mount point.
+        2) check disk name.
+        3) check disk's serial number.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+
+        """
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Check all file system info in a loop.", logging.info)
+        fs_info_qga = self.gagent.get_fsinfo()
+        for fs in fs_info_qga:
+            mount_pt = fs["mountpoint"]
+            if params["os_type"] == "windows":
+                mount_pt = mount_pt[:2]
+
+            error_context.context("Check file system of '%s' mount point." %
+                                  mount_pt, logging.info)
+            fs_qga = fs["type"]
+            cmd_get_disk = params["cmd_get_disk"] % mount_pt
+            disk_info_guest = session.cmd(cmd_get_disk).strip().split()
+            fs_guest = disk_info_guest[1]
+            if fs_qga != fs_guest:
+                test.fail("File System doesn't match.\n"
+                          "fs from guest-agent is %s,fs from guest os is %s."
+                          % (fs_qga, fs_guest))
+
+            error_context.context("Check disk name.", logging.info)
+            disk_name_qga = fs["name"]
+            disk_name_guest = disk_info_guest[0]
+            if params["os_type"] == "linux":
+                if not re.findall(r'^/\w*/\w*$', disk_name_guest):
+                    cmd = "ll %s |awk -F '/' '{print $5}' " % disk_name_guest
+                    disk_name_guest = session.cmd(cmd).strip()
+                else:
+                    disk_name_guest = disk_name_guest.split('/')[2]
```

Comment:
```
if not re.findall(r'^/\w*/\w*$', disk_name_guest):
    disk_name_guest = session.cmd("readlink %s" %disk_name_guest).strip()
disk_name_guest = disk_name_guest.split('/')[-1]
```

Reply:
@yanan-fu , Yes,readline is better. thanks.
Modified.

Reply:
@xiagao  I think you miss some info of the comment above,  all of these two scenario can use `[-1]`

Reply:
No, they are different. one is like /dev/sdx and another is ../dm-0

Reply:
You are right.Modified.